base_layer/core/src/block.rs

```diff
@@ -157,4 +166,135 @@ impl From<Transaction> for AggregateBody {
     }
 }
 
+#[derive(Default)]
+pub struct BlockBuilder {
+    pub header: Option<BlockHeader>,
+    pub inputs: Option<Vec<TransactionInput>>,
+    pub outputs: Option<Vec<TransactionOutput>>,
+    pub kernels: Option<Vec<TransactionKernel>>,
+}
+
+impl BlockBuilder {
+    pub fn new() -> BlockBuilder {
+        BlockBuilder {
+            header: None,
+            inputs: None,
+            outputs: None,
+            kernels: None,
+        }
+    }
+
+    /// This function adds a header to the block
+    pub fn with_header(&mut self, header: BlockHeader) -> &Self {
+        self.header = Some(header);
+        self
+    }
+
+    /// This function adds the provided transaction inputs to the block
+    pub fn with_inputs(mut self, inputs: &mut Vec<TransactionInput>) -> Self {
```

Comment:
`with_xxx` suggests that the value will be replaced by the input argument. `add_xxx` feels more natural since you're extending the Vec, noit replacing it.